Application.hpp

```diff
@@ -0,0 +1,21 @@
+#ifndef APPLICATION_H_
+#define APPLICATION_H_
+
+namespace spic
+{
+	class Application
+	{
+	public:
+		Application();
+		virtual ~Application();
+
+		void Run();
+		virtual void OnUpdate();
+
+	protected:
+	#include "Application_protected.hpp"

```

Comment:
Is hier sprake van templating? Protected variabelen kunnen anders gewoon de API in toch?

Reply:
Dit is gedaan volgens de richtlijnen van school zie [het document](https://brightspace.avans.nl/d2l/le/lessons/38377/topics/332939)

Reply:
Volgens dat document op brightspace moet er ook een header komen voor protected en ik ben het hier ook mee eens, aangezien dit nog steeds een soort van private variabelen zijn die specifiek kunnen zijn voor iedere groep

Reply:
Ik vind het prima, dit levert idd ook flexibiliteit op, thanks voor de uitleg.